src/main/java/edu/mit/puzzle/cube/core/model/User.java

```diff
@@ -0,0 +1,26 @@
+package edu.mit.puzzle.cube.core.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.auto.value.AutoValue;
+
+import java.util.List;
+
+@AutoValue
+@JsonDeserialize(builder = AutoValue_User.Builder.class)
+public abstract class User {
+    @AutoValue.Builder
+    public static abstract class Builder {
+        @JsonProperty("username") public abstract Builder setUsername(String username);
+        @JsonProperty("permissions") public abstract Builder setPermissions(List<String> permissions);
+
+        public abstract User build();
+    }
+
+    public static Builder builder() {
+        return new AutoValue_User.Builder();
+    }
+
+    @JsonProperty("username") public abstract String getUsername();
+    @JsonProperty("permissions") public abstract List<String> getPermissions();
```

Comment:
For those sorts of decisions, it might be preferable to define an /authorized route to make the decisions within the service, or what's the point of having an authorization engine. If it's one or two UI elements then it probably doesn't matter too much. As the UI decision points grow more complicated, we'd end up replicating the authorization logic in both places.